src/main/java/org/truffleruby/language/arguments/CheckKeywordArgumentNode.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2021 Oracle and/or its affiliates. All rights reserved. This
+ * code is released under a tri EPL/GPL/LGPL license. You can use it,
+ * redistribute it and/or modify it under the terms of the:
+ *
+ * Eclipse Public License version 2.0, or
+ * GNU General Public License version 2, or
+ * GNU Lesser General Public License version 2.1.
+ */
+package org.truffleruby.language.arguments;
+
+import com.oracle.truffle.api.frame.FrameSlot;
+import com.oracle.truffle.api.frame.VirtualFrame;
+import org.truffleruby.core.hash.RubyHash;
+import org.truffleruby.core.hash.library.HashStoreLibrary;
+import org.truffleruby.core.symbol.RubySymbol;
+import org.truffleruby.language.RubyContextSourceNode;
+import org.truffleruby.language.RubyNode;
+import org.truffleruby.language.arguments.keywords.KeywordDescriptor;
+import org.truffleruby.language.arguments.keywords.NonEmptyKeywordDescriptor;
+import org.truffleruby.language.locals.WriteLocalVariableNode;
+
+public class CheckKeywordArgumentNode extends RubyContextSourceNode {
```

Comment:
This is called CheckKeywordArgumentNode but it doesn't seem to do that.
What it does is to execute default value for kwargs (if the value is not already set), right?
So maybe `AssignKeywordDefaultValueNode`/`KeywordDefaultValueIfUnsetNode` or so.